.circleci/config.yml

```diff
@@ -432,6 +428,13 @@ jobs:
     parallelism: 6
     steps:
       - checkout
+      - run:
+          name: Halt if no cucumber tests to run
+          command: |
+            mkdir -p ./tmp && \
+            >./tmp/tests.txt && \
+            circleci tests glob "features/**/*.feature" | circleci tests run --command ">./tmp/tests.txt xargs echo" --split-by=timings
```

Comment:
not an issue, just that xargs defaults to `echo` so maybe echo isn't needed here? 

i'm learning as I read this PR .

Reply:
Both good questions @Obsiye. To be honest, I didn't write that command myself, I just copied it from the CircleCI guidance: https://circleci.com/docs/rerun-failed-tests/#known-limitations 😀 

I think you're right about `echo` not being needed. I'll test it when I'm back from leave.

My shell scripting is basic, but as I understand it, the `>` in `>./tmp/tests.txt` is a redirect, so that command taken as a whole creates the `./tmp` directory, then creates the file `tests.txt` in it, and redirects the output of the following command into that file. The third line runs the code which splits the tests between parallel containers, so we end up with a file in `./tmp/tests.txt` which contains the names of all of the tests which would be run in that container.

The next line `[ -s tmp/tests.txt ] || circleci-agent step halt` checks to see if there are any tests to be run, and if not, stops.

Hopefully that helps (but possibly didn't make any sense at all!) We can catch up on it in a couple of weeks if you want.